moo.js

```diff
@@ -170,7 +171,7 @@
       parts.push(reCapture(pat))
     }
 
-    var suffix = hasSticky ? '' : '|(?:)'
+    var suffix = hasSticky ? '' : '|'
```

Comment:
I can't remember what `(?:)` does, but is it definitely okay to remove this? What difference does it make?

Reply:
(?:) is an empty capture group, so it's identical to the empty regular expression. The reason it gets thrown around is because `RegExp.prototype.source` requires the result to parse as a regex, and `//` parses as a line comment. The spec-recommended workaround is `/(?:)/`.

Reply:
Oh, of course. Thanks for explaining 🙌